iroha/src/genesis.rs

```diff
@@ -0,0 +1,110 @@
+//! This module contains execution Genesis Block logic, and GenesisBlock definition.
+use crate::{init::config::InitConfiguration, tx::Accept, tx::AcceptedTransaction, Identifiable};
+use iroha_crypto::KeyPair;
+use iroha_data_model::{account::Account, isi::InstructionBox, prelude::*};
+use serde::Deserialize;
+use std::{fmt::Debug, fs::File, io::BufReader, path::Path};
+
+/// GenesisBlock contains transactions for inisialize settings.
+#[derive(Clone, Debug)]
+pub struct GenesisBlock {
+    /// transactions from GenesisBlock, any transaction is accepted
+    pub txs: Vec<AcceptedTransaction>,
+}
+
+#[derive(Clone, Deserialize, Debug)]
+struct RawGenesisBlock {
+    pub txs: Vec<GenesisTx>,
+}
+
+/// GenesisTx is a transaction for inisialize settings.
+#[derive(Clone, Deserialize, Debug)]
+struct GenesisTx {
```

Comment:
An idiomatic way to implement a conversion in rust would be to implement: `impl TryFrom<GenesisTx> for AcceptedTransaction` and use it in the `map` below.

Reply:
I agree with your idea, but I have a question.
Now, conversion logic requires keypair for signing....
so how to use TryFrom? I thought some method but not I'm not convinced.
1) GenesisTx has a keypair
2) Define PairType as follow
```
struct GenesisTxWithKey {
    txs: GenesisTx
    key: KeyPair
}

impl TryFrom<GenesisTxWithKey> for AcceptedTransaction
```

Reply:
Okay, I see. Yes you are right, implementing `TryFrom` will not be good here. Maybe you can do something like that:
```rust
impl GenesisTransaction {
    pub fn sign_and_accept(key_pair) -> AcceptedTransaction {
        ...
    }
}
```